includes/Core/Util/URL.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Util\URL
+ *
+ * @package   Google\Site_Kit\Core\Util
+ * @copyright 2022 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Util;
+
+/**
+ * Class for custom URL parsing methods.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class URL {
+
+	/**
+	 * Parses URLs with UTF-8 multi-byte characters,
+	 * otherwise similar to `wp_parse_url()`.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $url       The URL to parse.
+	 * @param int    $component The specific component to retrieve. Use one of the PHP
+	 *                          predefined constants to specify which one.
+	 *                          Defaults to -1 (= return all parts as an array).
+	 * @return mixed False on parse failure; Array of URL components on success;
+	 *               When a specific component has been requested: null if the component
+	 *               doesn't exist in the given URL; a string or - in the case of
+	 *               PHP_URL_PORT - integer when it does. See parse_url()'s return values.
+	 */
+	public static function parse( $url, $component = -1 ) {
+		if ( ! mb_strlen( $url, 'UTF-8' ) !== strlen( $url ) ) {
```

Comment:
We can just use `===`, can't we?
```suggestion
		if ( mb_strlen( $url, 'UTF-8' ) === strlen( $url ) ) {
```